src/baldr/shortcut_recovery.h

```diff
@@ -174,7 +178,9 @@ struct shortcut_recovery_t {
       }
 
       // if we didnt add an edge or we went over the length we failed
-      if (current_edge == nullptr || accumulated_length > shortcut->length()) {
+      ++edge_count;
+      roundoff_error = std::round(edge_count * 0.5) + 1;
+      if (current_edge == nullptr || accumulated_length > (shortcut->length() + roundoff_error)) {
```

Comment:
this check may not be needed. Accumulating edge lengths (which have been truncated to int) should never lead to being greater than the shortcut length computed from the shape (and then truncated to int).